kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -363,13 +420,50 @@ private SnapshotImpl getSnapshotAtInit(Engine engine)
             logger.warn("{}: Last checkpoint file is missing or corrupted. " +
                     "Will search for the checkpoint files directly.", tablePath);
         }
-        Optional<LogSegment> logSegmentOpt =
-            getLogSegmentFrom(engine, lastCheckpointOpt);
+        Optional<LogSegment> logSegmentOpt = getLogSegmentFrom(engine, lastCheckpointOpt);
 
         return logSegmentOpt
-            .map(logSegment -> createSnapshot(
-                logSegment, engine))
-            .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+                .map(logSegment -> getUpdatedSnapshot(
+                        engine,
+                        Optional.empty(), /* oldSnapshotOpt */
+                        logSegment,
+                        Optional.empty() /* initialCommitCoordinatorClientHandler */))
+                .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+    }
+
+    private SnapshotImpl getUpdatedSnapshot(
```

Comment:
Kernel does not have the concept of a post-commit snapshot, so I don't think that this is useful? So, kernel will not have an `oldSnapshotOpt` nor an `initialCommitCoordinatorClientHandler`, the logic can be simplified a lot.